src/swap/defi/uni-v2-based/uniV2Utils.js

```diff
@@ -46,6 +46,7 @@ export const getSwapAmounts = async (
  * Get smart contract transaction(s) necessary to swap based on swap params
  */
 export const getSwapTransactions = async (
+  provider: JsonRpcProvider,
```

Comment:
These functions feel like methods to the swap plugin rather than utils; don't you think?

Reply:
I'll secede this request for a later refactor.